pkg/synchronizer/kubernetes/synchronizer.go

```diff
@@ -325,6 +332,17 @@ func (sync *KubeSynchronizer) ProcessSubResources(appsub *appv1alpha1.Subscripti
 		appSubUnitStatuses = append(appSubUnitStatuses, appSubUnitStatus)
 	}
 
+	endTime := time.Now().UnixMilli()
+	if gotDeployErrs {
+		metrics.LocalDeploymentFailedPullTime.
+			WithLabelValues(appsub.Namespace, appsub.Name).
+			Observe(float64(endTime - startTime))
+	} else {
+		metrics.LocalDeploymentSuccessfulPullTime.
+			WithLabelValues(appsub.Namespace, appsub.Name).
+			Observe(float64(endTime - startTime))
+	}
+
 	appsubClusterStatus := SubscriptionClusterStatus{
```

Comment:
The end time should be given after this call
```
err := sync.SyncAppsubClusterStatus(appsub, appsubClusterStatus, nil, nil)
```
This is for recording the appsub package status to the appsubStatus CR on the managed cluster and the cluster level appsubReport in the managed cluster namespace on the hub. It is part of local deployment.

Reply:
fixed in the latest push.